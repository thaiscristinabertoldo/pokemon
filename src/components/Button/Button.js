import React, { memo } from 'react'

import { ButtonStyled, ButtonOutlineStyled } from './button.css'

export const Button = memo(({ text, prefix, onClick, outline, ...props }) => {
  return (
    <>
      {outline ? (
        <ButtonOutlineStyled onClick={onClick} {...props}>
          {prefix} {text}
        </ButtonOutlineStyled>
      ) : (
        <ButtonStyled onClick={onClick} data-testid="button" {...props}>
          {prefix} {text}
        </ButtonStyled>
      )}
    </>
  )
})
