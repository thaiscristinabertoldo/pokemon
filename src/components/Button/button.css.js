import styled from 'styled-components'

import { COLORS } from '../../constants'

const colorDisabled = COLORS.lightGrey

const ButtonStyled = styled.button`
  padding: 10px;
  border: 1px solid ${(prop) => (prop.disabled ? colorDisabled : prop.backgroundColor)};
  background-color: ${(prop) => (prop.disabled ? colorDisabled : prop.backgroundColor)};
  border-radius: 5px;
  color: ${(prop) => prop.color};
  font-weight: bold;
  font-size: 16px;
  box-shadow: 2px 2px 6px ${colorDisabled};
  width: ${(prop) => prop.width};
  cursor: ${(prop) => (prop.disabled ? 'default' : 'pointer')};
  &:focus {
    outline: none;
  }
`

ButtonStyled.defaultProps = {
  backgroundColor: COLORS.secondary,
  color: COLORS.white,
  width: 'auto',
}

const ButtonOutlineStyled = styled.button`
  padding: 10px;
  border: 1px solid ${(prop) => (prop.disabled ? colorDisabled : prop.backgroundColor)};
  background-color: ${COLORS.white};
  border-radius: 5px;
  color: ${(prop) => (prop.disabled ? colorDisabled : prop.backgroundColor)};
  font-weight: bold;
  font-size: 16px;
  box-shadow: 2px 2px 6px ${colorDisabled};
  width: ${(prop) => prop.width};
  cursor: ${(prop) => (prop.disabled ? 'default' : 'pointer')};
  &:focus {
    outline: none;
  }
`

ButtonOutlineStyled.defaultProps = {
  backgroundColor: COLORS.secondary,
  width: 'auto',
}

export { ButtonStyled, ButtonOutlineStyled }
