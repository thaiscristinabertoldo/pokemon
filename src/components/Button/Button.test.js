import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Button } from './Button'

afterEach(cleanup)

describe('Button', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Button />)

    expect(asFragment(<Button />)).toMatchSnapshot()
  })

  it('should equal to Text button', () => {
    const { getByText } = render(<Button text="Text button" />)
    expect(getByText('Text button')).toBeTruthy()
  })

  it('should be enabled', () => {
    const { getByTestId } = render(<Button text="Text button" />)
    expect(getByTestId('button')).not.toHaveAttribute('disabled')
  })

  it('should be disabled', () => {
    const { getByTestId } = render(<Button text="Text button" disabled />)
    expect(getByTestId('button')).toHaveAttribute('disabled')
  })
})
