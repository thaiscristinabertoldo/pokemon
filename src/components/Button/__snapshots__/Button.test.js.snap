// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm ikEyMv"
    color="#ffffff"
    data-testid="button"
    width="auto"
  >
     
  </button>
</DocumentFragment>
`;
